src/xia2/Modules/MultiCrystal/__init__.py

```diff
@@ -1,478 +0,0 @@
-from scitbx.array_family import flex
-
-logger = logging.getLogger(__name__)
-
-batch_phil_scope = """\
-batch
-  .multiple = True
-{
-  id = None
-    .type = str
-  range = None
-    .type = ints(size=2, value_min=0)
-}
-"""
-
-master_phil_scope = iotbx.phil.parse(
-    """\
-unit_cell = None
-  .type = unit_cell
-  .short_caption = "Unit cell"
-n_bins = 20
-  .type = int(value_min=1)
-  .short_caption = "Number of bins"
-d_min = None
-  .type = float(value_min=0)
-  .short_caption = "High resolution cutoff"
-%s
-"""
-    % batch_phil_scope
-)
-
-
-class ClusterInfo:
-    def __init__(
-        self, cluster_id, labels, multiplicity, completeness, unit_cell, height=None
-    ):
-        self.cluster_id = cluster_id
-        self.labels = labels
-        self.multiplicity = multiplicity
-        self.completeness = completeness
-        self.unit_cell = unit_cell
-        self.height = height
-
-    def __str__(self):
-        lines = [
-            "Cluster %i" % self.cluster_id,
-            "  Number of datasets: %i" % len(self.labels),
-            "  Completeness: %.1f %%" % (self.completeness * 100),
-            "  Multiplicity: %.2f" % self.multiplicity,
-            "  Datasets:" + ",".join("%s" % s for s in self.labels),
-        ]
-        if self.height is not None:
-            lines.append("  height: %f" % self.height)
-        return "\n".join(lines)
-
-
-class multi_crystal_analysis:
-    def __init__(self, unmerged_intensities, labels=None, prefix=None):
-        self.unmerged_intensities = unmerged_intensities
-        self._intensities_all = None
-        self._labels_all = flex.size_t()
-        if prefix is None:
-            prefix = ""
-        self._prefix = prefix
-
-        self.intensities = unmerged_intensities
-        self.individual_merged_intensities = []
-        if labels is None:
-            labels = ["%i" % (i + 1) for i in range(len(self.intensities))]
-        assert len(labels) == len(self.intensities)
-        self.labels = labels
-
-        for i, unmerged in enumerate(self.intensities):
-            self.individual_merged_intensities.append(
-                unmerged.merge_equivalents().array().set_info(unmerged.info())
-            )
-            if self._intensities_all is None:
-                self._intensities_all = unmerged.deep_copy()
-            else:
-                self._intensities_all = self._intensities_all.concatenate(
-                    unmerged, assert_is_similar_symmetry=False
-                )
-            self._labels_all.extend(flex.size_t(unmerged.size(), i))
-
-        self.run_cosym()
-        (
-            correlation_matrix,
-            linkage_matrix,
-        ) = self.compute_correlation_coefficient_matrix()
-
-        cos_angle_matrix, ca_linkage_matrix = self.compute_cos_angle_matrix()
-
-        d = self.to_plotly_json(correlation_matrix, linkage_matrix, labels=labels)
-
-        with open("%sintensity_clusters.json" % self._prefix, "w") as f:
```

Comment:
We are now missing this in the output files, so need to add back into `cluster_analysis.py` somewhere? Also was this output by multiplex too?

Reply:
Have added output of cluster.json files for both `cluster_analysis.py` and `multiplex` - found a bug though when running the multiplex tests - have fixed and made a PR in dials to address this. Will need to be merged so these tests can be rerun.

Reply:
I had misunderstood that this was not quite the same as what we are outputting now - the data in these json files were the plotly data, which is also contained in `xia2.multiplex.json`. But it's good to have the new output json which can be useful for follow on analysis 🙂